test/jdk/sun/tools/jmap/BasicJMapTest.java

```diff
@@ -237,25 +246,42 @@ private static void dump(boolean live, boolean explicitAll) throws Exception {
             liveArg = "all,";
         }
 
-        File file = new File("jmap.dump" + System.currentTimeMillis() + ".hprof");
+        String filePath = "jmap.dump" + System.currentTimeMillis() + ".hprof";
+        if (compressed) {
+            compressArg = "gz=1,";
+            filePath = filePath + ".gz";
+        }
+
+        File file = new File(filePath);
         if (file.exists()) {
             file.delete();
         }
         fileArg = "file=" + file.getName();
 
         OutputAnalyzer output;
-        allArgs = allArgs + liveArg + "format=b," + fileArg;
+        allArgs = allArgs + liveArg + compressArg + "format=b," + fileArg;
         output = jmap(allArgs);
         output.shouldHaveExitValue(0);
         output.shouldContain("Heap dump file created");
-        verifyDumpFile(file);
+        verifyDumpFile(file, compressed);
         file.delete();
     }
 
-    private static void verifyDumpFile(File dump) {
+    private static void verifyDumpFile(File dump, boolean compressed) {
```

Comment:
I don't think the `compressed` argument is needed. The rest of the changes look fine.